add latency tracer support for EP93xx boards


Add latency tracer support for the EP93xx platform. This is done by:
- adding the correct Kconfig options
- add (an empty) save_stack_trace implementation.
  -> Someone needs to implement save_stack_trace for arm :)
     Maybe we can use the implementation from rmk?
- implementing mach_read_cycles (read out EP93XX_TIMER4_VALUE_LOW)
- implementing mach_cycles_to_usecs (just the same way as for the PXA platform)
- implementing mach_usecs_to_cycles (just the same way as for the PXA platform)

Signed-off-by: Jan Altenberg <jan@linutronix.de>
---
 arch/arm/Kconfig                    |    4 ++++
 arch/arm/lib/Makefile               |    1 +
 arch/arm/lib/stacktrace.c           |    7 +++++++
 include/asm-arm/arch-ep93xx/timex.h |    6 ++++++
 4 files changed, 18 insertions(+)

Index: linux-2.6.23.1-rt5/arch/arm/Kconfig
===================================================================
--- linux-2.6.23.1-rt5.orig/arch/arm/Kconfig
+++ linux-2.6.23.1-rt5/arch/arm/Kconfig
@@ -33,6 +33,10 @@ config GENERIC_CLOCKEVENTS
 	bool
 	default n
 
+config STACKTRACE_SUPPORT
+	bool
+	default y
+
 config MMU
 	bool
 	default y
Index: linux-2.6.23.1-rt5/arch/arm/lib/Makefile
===================================================================
--- linux-2.6.23.1-rt5.orig/arch/arm/lib/Makefile
+++ linux-2.6.23.1-rt5/arch/arm/lib/Makefile
@@ -41,6 +41,7 @@ lib-$(CONFIG_ARCH_RPC)		+= ecard.o io-ac
 lib-$(CONFIG_ARCH_CLPS7500)	+= io-acorn.o
 lib-$(CONFIG_ARCH_L7200)	+= io-acorn.o
 lib-$(CONFIG_ARCH_SHARK)	+= io-shark.o
+lib-$(CONFIG_STACKTRACE)	+= stacktrace.o
 
 $(obj)/csumpartialcopy.o:	$(obj)/csumpartialcopygeneric.S
 $(obj)/csumpartialcopyuser.o:	$(obj)/csumpartialcopygeneric.S
Index: linux-2.6.23.1-rt5/arch/arm/lib/stacktrace.c
===================================================================
--- /dev/null
+++ linux-2.6.23.1-rt5/arch/arm/lib/stacktrace.c
@@ -0,0 +1,7 @@
+#include <linux/sched.h>
+#include <linux/stacktrace.h>
+
+void save_stack_trace(struct stack_trace *trace)
+{
+}
+
Index: linux-2.6.23.1-rt5/include/asm-arm/arch-ep93xx/timex.h
===================================================================
--- linux-2.6.23.1-rt5.orig/include/asm-arm/arch-ep93xx/timex.h
+++ linux-2.6.23.1-rt5/include/asm-arm/arch-ep93xx/timex.h
@@ -1,5 +1,11 @@
 /*
  * linux/include/asm-arm/arch-ep93xx/timex.h
  */
+#include <asm-arm/arch-ep93xx/ep93xx-regs.h>
+#include <asm-arm/io.h>
 
 #define CLOCK_TICK_RATE		983040
+
+#define mach_read_cycles() __raw_readl(EP93XX_TIMER4_VALUE_LOW)
+#define mach_cycles_to_usecs(d) (((d) * ((1000000LL << 32) / CLOCK_TICK_RATE)) >> 32)
+#define mach_usecs_to_cycles(d) (((d) * (((long long)CLOCK_TICK_RATE << 32) / 1000000)) >> 32)
